doc/Caliptra.md

```diff
@@ -1226,20 +1248,199 @@ This section describes Caliptra error reporting and handling.
+6. MCU HW logic will use this indication to initiate MCU uC reset flow
+   a. MCU HW logic sends a reset-go-req to MCU uC (an interrupt)
+   b. MCU HW logic waits for reset-go-ack from MCU uC (Note that this handshake exists to ensure uController is in an appropriate quiescent state to take the reset)
+   c. MCU HW logic will assert the reset to the MCU uC
+7. Caliptra RT FW will wait for reset assertion and then read the staged SRAM over AXI manager interface and write the image to the MCU SRAM aperture (that is open to Caliptra only by HW construction).
+   a. The address of the MCU SRAM is provided to Caliptra’s RT FW through SOC manifest.
+   b. Note: From the validation front, need to ensure the Caliptra ROM and MCU are aligned in endianness.
+   c. Note: True downtime of MCU is from when its reset is asserted; It is SOC implementation requirement that it handles (eg. through buffering) all transactions to MCU while it is going through a hitless update.
+8. After the MCU SRAM is populated, Caliptra RT FW will set EXEC/GO bit into the register as specified in the previous command. This register write will also assert a Caliptra interface wire.
+9. MCU HW logic will use this indication to deassert the MCU reset.
+10. MCU ROM will look at the breadcrumb that the MCU SRAM has valid content and will start the execution from it directly. **NOTE:** This breadcrumb should be one of the status bits available on the MCU interface that is set by Caliptra GO bit wires.
+
+**SOC-FW Hitless Update**
+
+SOC may have other components that may need to be updated at run-time in a hitless/impactless manner.
+
+The update flow will follow the same sequence as MCU Hitless update except they are executed by the MCU by using Caliptra as the RoT engine for doing all the required authentication/authorization flows.
+
+Further SOCs may require the hitless update without impacting the workloads/VMs running on the host or the VMs using the devices. This essentially means that impactless update must happen without causing any timeouts to the in-flight transactions. While the treatment of those transactions are device dependent, Caliptra subsystem must provide a way to be able to authenticate and activate the FW in the shortest time possible.
+
+Caliptra subsystem provides this architectural capability as follows:
+1. MCU provides SOC manifest to Caliptra and waits for authentication to be successful. If this wasn’t provided Caliptra will use the latest SOC manifest available.
+   a. If failed, MCU uses MCTP PLDM to report the same to the update agent using PLDM protocol
+2. MCU stages all the incoming FW payload in an SOC-defined staging memory and provides the MMIO address of the staging SRAM to Caliptra. It is better to keep this as a part of the authenticated SOC manifest (as a configuration) from security perspective.
```

Comment:
should state this as a requirement to keep the MMIO address in the manifest

Reply:
Allow FW arch team to set that manifest or should we just make it a tops down requirement?

Reply:
top down because it's required by the threat model

Reply:
Added this - "(**FW Arch requirement:** This SOC-defined staging RAM MMIO offset which can be one for all the images or it could be per image, recommended to keep it the later way, should be defined in the SOC manifest.)"